app/controllers/editions_controller.rb

```diff
@@ -2,39 +2,35 @@
 
 class EditionsController < ApplicationController
   def create
-    Document.transaction do
-      document = Document.with_current_edition.lock.find_by_param(params[:document_id])
-
-      current_edition = document.current_edition
-
-      unless current_edition.live?
+    Edition.find_and_lock_current(document: params[:document]) do |edition|
```

Comment:
Just to confirm, we don't need to lock on the document here, as we'll be updating the `current` property of the edition we're locking on?

Reply:
We lock on the document and the edition with `find_and_lock_current` method so both bases are covered, but I think you're right that we'd get away with just a lock on edition here anyway.